docs/pubsubsource/README.md

```diff
@@ -0,0 +1,207 @@
+[Cloud Storage](https://cloud.google.com/storage/docs/pubsub-notifications),
+[IoT Core](https://cloud.google.com/iot/docs/how-tos/devices) and
+[Cloud Scheduler](https://cloud.google.com/scheduler/docs/creating#).
+
+## Prerequisites
+
+1. Create a
+   [Google Cloud project](https://cloud.google.com/resource-manager/docs/creating-managing-projects)
+   and install the `gcloud` CLI and run `gcloud auth login`. This sample will
+   use a mix of `gcloud` and `kubectl` commands. The rest of the sample assumes
+   that you've set the `$PROJECT_ID` environment variable to your Google Cloud
+   project id, and also set your project ID as default using
+   `gcloud config set project $PROJECT_ID`.
+
+1. [Install Cloud Run Events](../install).
+
+1. Enable the `Cloud Pub/Sub API` on your project:
+
+   ```shell
+   gcloud services enable pubsub.googleapis.com
+   ```
+
+1. Create a
+   [GCP Service Account](https://console.cloud.google.com/iam-admin/serviceaccounts/project).
+   This sample creates one service account for both registration and receiving
+   messages, but you can also create a separate service account for receiving
+   messages if you want additional privilege separation.
+
+   1. Create a new service account named `cloudrunevents-source` with the
+      following command:
+      ```shell
+      gcloud iam service-accounts create cloudrunevents-source
+      ```
+   1. Give that Service Account the `Pub/Sub Editor` role on your GCP project:
+      ```shell
+      gcloud projects add-iam-policy-binding $PROJECT_ID \
+        --member=serviceAccount:cloudrunevents-source@$PROJECT_ID.iam.gserviceaccount.com \
+        --role roles/pubsub.editor
+      ```
+   1. Download a new JSON private key for that Service Account. **Be sure not to
+      check this key into source control!**
+      ```shell
+      gcloud iam service-accounts keys create cloudrunevents-source.json \
+        --iam-account=cloudrunevents-source@$PROJECT_ID.iam.gserviceaccount.com
+      ```
+   1. Create two secrets on the kubernetes cluster with the downloaded key:
+
+      ```shell
+      # Note that the first secret may already have been created when installing
+      # Cloud Run Events. The following command will overwrite it. If you don't
+      # want to overwrite it, then skip this command.
+      kubectl --namespace cloud-run-events create secret generic cloud-run-events-source-key --from-file=key.json=cloudrunevents-source.json --dry-run --output yaml | kubectl apply --filename -
+
+      # The second secret should not already exist, so just try to create it.
+      kubectl --namespace default create secret generic google-cloud-key --from-file=key.json=cloudrunevents-source.json
+      ```
+
+      `cloud-run-events-source-key` and `key.json` are pre-configured values in
+      the `controller` Deployment which manages Cloud Run Events resources.
+
+      `google-cloud-key` and `key.json` are pre-configured values in
```

Comment:
so, this is the same as before? Is not multi-tenant then? The controller is created for a single project?

Reply:
Updated to be multi-tenant, with a TODO to clean up subscriptions.